packages/ibm-products-web-components/src/components/truncated-text/truncated-text.ts

```diff
@@ -0,0 +1,214 @@
+    }
+    requestAnimationFrame(() => {
+      if (!this._textElement) {
+        return;
+      }
+      this._maxHeight =
+        this.lines > 0 && !this._isExpanded
+          ? `${this.lines * this._lineHeight}px`
+          : `${this._textElement.scrollHeight}px`;
+    });
+  }
+
+  private _setupResizeObserver() {
+    if (!this._textElement) {
+      return;
+    }
+
+    if (this._resizeObserver) {
+      this._resizeObserver.disconnect();
+    }
+
+    this._resizeObserver = new ResizeObserver(() => {
+      this._updateOverflowStatus();
+    });
+
+    this._resizeObserver.observe(this);
+  }
+
+  private _updateOverflowStatus() {
+    if (!this._textElement || this.lines <= 0) {
+      return;
+    }
+    this._updateMaxHeight();
+    const { scrollHeight, clientHeight } = this._textElement;
+    const buffer = this._lineHeight / 2; // buffer of at least half of line height for a stable outcome
+
+    const isOverflowing = scrollHeight > clientHeight + buffer;
+
+    if (isOverflowing !== this._isOverflowing) {
+      this._isOverflowing = isOverflowing;
+    }
+  }
+
+  private _toggleExpansion() {
+    this._isExpanded = !this._isExpanded;
+    this._updateMaxHeight();
+    this._textElement?.classList.add(`${blockClass}_transition`);
+    const onTransitionEnd = () => {
+      this._textElement?.querySelector('button')?.focus();
+      this._textElement?.removeEventListener('transitionend', onTransitionEnd);
+    };
+    this._textElement?.addEventListener('transitionend', onTransitionEnd);
+  }
+
+  private _renderToggleButton() {
+    if (this.with !== 'expand') {
+      return;
+    }
+    const className = classMap({
+      [`${blockClass}_button-collapse`]: this._isExpanded,
+      [`${blockClass}_button-expand`]: !this._isExpanded,
+      [`${blockClass}_button-layered`]: this._isLayered,
+      [`${blockClass}_button-hide`]: !this._isOverflowing && !this._isExpanded,
+    });
+    const label = this._isExpanded
+      ? this.collapseLabel || '...less'
+      : this.expandLabel || '...more';
+    return html`
+      <button class="${className}" @click=${this._toggleExpansion}>
+        ${label}
+      </button>
+    `;
+  }
+
+  render() {
+    // Apply different styles based on truncation method
+    const contentStyle =
+      this.with === 'tooltip' || !this.with
+        ? `--line-clamp: ${this._isExpanded ? 'none' : this.lines};`
+        : `max-block-size: ${this._maxHeight};`;
+
+    const contentClass = classMap({
+      [`${blockClass}_content`]: true,
+    });
+
+    const content = html`
+      <div
+        class="${contentClass}"
+        style="${contentStyle}"
+        tabindex=${this.with === 'tooltip' && this._isOverflowing
+          ? '0'
+          : undefined}
+      >
+        ${this.value} ${this._renderToggleButton()}
+      </div>
+    `;
+
+    return this.with === 'tooltip' && this._isOverflowing
+      ? html`
+          <cds-tooltip align="bottom" enter-delay-ms="0" leave-delay-ms="0">
```

Comment:
I feel like we should support both `autoAlign` and a manual `tooltipAlign` when using the tooltip.